### DISCUSSION TITLE
Final Skill Revamp Proposal (FINALDRAFT_v3.1_FINALFINAL(?)_v420)

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#3419] [Ideas]
Below is my final Skill Tree Overhaul proposal aiming to vastly improve the amount of skills players can choose in the game whilst also providing incentives for players to level up. Additionally, the large variety in skills will help to diversify players' farms depending on which section of the gameplay they wish to develop into. 

Prior proposals:
[Skill tree revamp](https://github.com/sunflower-land/sunflower-land/discussions/2180)
[Skill tree revamp 2](https://github.com/sunflower-land/sunflower-land/discussions/3202)

### The basics and set up
- System to be launched with or soon after April 1st Bumpkin locking/withdrawals disabled.
- 1 level = 1 skill point, therefore everytime a player levels up they receive 1 skill point.
- Upon introduction of the new skill system, players will automatically have their current skills and skill points removed. Players will then automatically be given the amount of skill points related to their Bumpkin Level. EG: If a player has a level 56 bumpkin, they will receive 56 skill points. 1 point per level.
- Players currently will have their levels capped depending on which island they are at. Petal Island for example the cap will be at 55, any xp added on after that will not increase the level of the player and therefore not get more skill points. They will need to upgrade their land so that the level cap increases, allowing players to unlock more skill points. 
- Addition of future 'reset skills' ability. If the player feels they picked the 'wrong' skills or decide to change their gameplay or farm style, we could add a skill reset option that costs SFL, resources, BBs or a combination of all (leaving this open to be determined). A cooldown period would also be good to have, 3-6 months to prevent people to reset their skills frequently and pick skills that are hot with the market.

### The Skills:
_Final values and/or skills may change at release, this is my proposal for them at this stage._
_Blue text = trade-off skills, Red text = 'Actions'_
_Click on image to enlarge_
![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/0d756a66-fd7d-4d7e-8284-9b953ed338f3)

### Tier levels
Players can unlock as many tier 1 skills as they wish, if they have enough skill points. To unlock x many tier 2 skills in a tree, the player has to unlock the same x amount of the previous tier.
EG: To unlock 2 x Tier 2 skills in the crop three, a player has to unlock at least 2 x Tier 1 skills. The same applied for Tier 3. To unlock 4 x Tier 3 skills in a single tree, a player needs to unlock 4 x Tier 2 skills in the same tree, which also requires unlocking 4 x Tier 1 skills (of that same tree).
Fairly simple process and UI can help ingame. Selecting a Tier 2 skill without having Tier 1 unlocked could show: "x Tier 1 skills needed"

### End goal & progression
In my first post made over a year ago, I said the following: 
_"...would give the players more options and paths to how they want to play the game. Not only would it add variety and a bit of extra decision-making on behalf of the player, **but it would give a big incentive for players to level up to gain more skill points.**
At level 40 a player has pretty much covered most of the skills present/they wanted, **whilst these new skill trees would drive the established/maxed out players to progress further and have something to aim for.**"_

A big reason I want this to come to fruition is because I, as a player, WANT to have long-term goals to aim for and this would help me to give further meaning in levelling up, something which I often complained about (Just look above).

### More skills in future?
Yes, this system would allow for new skills to be added to existing trees or completely new trees altogether. Cows could have their own separate new tree or a new crop could have a skill introduced in the existing crop tree.
The only complexity is the need to have a minimum number of lower tiered skills to allow players to unlock higher tier skills.
EG: If cooking skill tree has 5 x Tier 3 skills but only 3 x Tier 2 skills, players won't be able to pick 2 of the 5 Tier 3 skills (unless more Tier 2 added).

### When?
If the team is happy and finds time, soonTM :D
The withdraw/deposit ability of bumpkins made things a lot trickier to balance and prevented a lot of ideas from being used. As bumpkins will soon be married to the farm, so will xp and skills which will make the process and system much easier to manage. I would dare say it should be implemented after this occurs (April 1st) as a bare minimum. Heavy testing will be required (Need to account for seasonal activities too, if/how it will impact)

### Final comments
Still happy to receive feedback but I do not plan to change the system in a big way, this will be my last post/proposal on skills. I have hopefully laid everything out for the team and community to follow on, including details and questions around specific scenarios.

I believe this will improve our gameplay and open up the diversity of farms. I also believe this will add further meaning to levelling up alongside the Island Upgrade feature adding incentives to food and xp.

If you notice any mistakes or glaring issues with some of the skills please feel free to comment. Another reminder these skills are not final and may/can be changed before release, particularly the values.

Thank you

**Comment by brian-man** [comment]
> Players currently will have their levels capped depending on which island they are at. Petal Island for example the cap will be at 55, any xp added on after that will not increase the level of the player and therefore not get more skill points. They will need to upgrade their land so that the level cap increases, allowing players to unlock more skill points.

I'm not liking this trend of taking away what people already earned.  If this is the model intended, why not wait for the new skills until L65 is available (with next island)?

**Reply by TheDcol** [reply]
Sorry i should have added that current players at level > 55 shouldnt be capped at release, they retain that high level.
But someone at < lvl 55 wouldnt unlock new levels.

Or the cleaner approach is to release with new island upgrade with a higher cap

**Comment by brian-man** [comment]
> To unlock x many tier 2 skills in a tree, the player has to unlock the same x amount of the previous tier.

This model doesn't *require* that higher tiers have the same or fewer than lower tiers, but it sure seems appropriate for that to be the case.

**Reply by TheDcol** [reply]
Yes, very appropriate.
Although, if there are fewer lower tiers than higher tiers the player would need to decide which higher tier to unlock.. IF.

**Comment by brian-man** [comment]
I see a few examples where I'm immediately curious if the skill is affected by collectibles with boost.  Bee Swarm being a good example.  Would be good to be clear in the phrasing.

**Reply by TheDcol** [reply]
Are you referring to 'guaranteed bee swarms' combined with either hornet mask and/or queen bee?
If so, they would stack. Granted hornet mask becomes a bit useless since 'guaranteed bee swarms' is 100% chance of happening..
With Honey Bee & guaranteed swarms, honey would collect in 24 hours BUT give guaranteed swarms. 24h / 2 + 12 hours

What i have isn't the final product, but hopefully is pretty damn close so we can have it ingame. We can definitely perfect the wording OR work out and solve any clear issues :)

**Reply by brian-man** [reply]
I was referring to honey production being slowed by 12hr when that's a 50% slowdown for those without Queen Bee but 100% slowdown for those with QB.

**Reply by TheDcol** [reply]
The 12 hour slowdown in the skill is a flat value

**Comment by adamhannigan** [comment]
Nice work @TheDcol 👏

- I love the 'mega' actions (e.g. instant grow crops) - these feel like a great reward for players who have climbed a progression tree. I would consider having these as a Tier 4 skill. Players would need to unlock all skills to unlock it. This would feel extremely rewarding to reach. I would even suggest having one action for each category

- The Tier system feels good, having to unlock X previous skills before unlocking a skill in a new tier. It also works well if we want to add new skills to a tier (a tree structure makes this hard)

- Similar to crops, I would suggest creating groups for the fruits instead of individual boosts for each fruit. In the future we will add more fruits. Consider (+0.1 for basic fruits - bluberry, orange, apple) and (+0.1 for advanced fruits - banana)

- It feels like we have too many chicken related boosts. I feel we should halve this amount and have a generic 'animals' category. Here we can include future cow boosts if needed.

Interested to hear more from the community about what skills + buffs they would like to see in the game ❤️ Awesome job mate!

**Reply by TheDcol** [reply]
Thanks Adam, here is my response:

- Would this be a bit tricky to manage or confusing to players if we were to add future skills to a skill tree that a player has fully completed? Would they then lose the Tier 4 action ability?

- Created groups for fruits as suggested

- Trimmed chicken skills and added Action skill to it. Also rename to ANIMALS

![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/89bec1e9-7e95-4210-ae8f-e50d64e279dc)
![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/93ad4df1-b11c-4cf9-8fff-e70c81b528d0)

**Reply by TheDcol** [reply]
Update, currently in planning

![skillsv5](https://github.com/sunflower-land/sunflower-land/assets/108110065/1066e40e-bf55-4667-b48f-1ce70baa6a89)

**Reply by heatedlime** [reply]
I think a cool one for fishing would be

“10% chance you don’t lose the bait”

**Comment by Dionis404** [comment]
Maybe we should add unique faction skills?
There will be some distinction in this and players will be able to choose a faction to suit their play style.

**Comment by ma-aguiar** [comment]
One idea would be to give free wearables (with no buffs) for each branch that the player completes. Those wearables would only serve to identify that this player is specialized in a certain area of the game.  For example, a florist shirt for someone that completes the entire Skill Set for Bees & Flowers.

**Reply by heatedlime** [reply]
What happens when they reset the skill and no longer have the complete branch completed?

**Reply by ma-aguiar** [reply]
> What happens when they reset the skill and no longer have the complete branch completed?

I haven't thought of that. Maybe the wearables get locked if the player resets. Or discard the wearables idea, and implement another feature to identify players that are specialized, like a [title].

**Reply by heatedlime** [reply]
Seems like a lot of extra coding work just for a wearable. I do like the idea of some sort of perk for being a master in one branch. Maybe there’s one skill in each branch that can only be unlocked if you’ve unlocked all other skills in that branch?

**Reply by Rudolph-the-Bumpkin** [reply]
I really like the idea. It seems like a lot of coding just for a wereable because, unfortunately, the game doesn't incentivize players to show off. 

I hope we do have more initiatives like we had in past couple streams, when we had "trivia show off" and "fashion show". This way people will actually like to show their exclusive titles, specialization etc which could justify this kind of development. 

A new skill would be cool, but I am kind of tired of the game just emphasizing utility aspects. Now that we're going to have 9 beds, we will finally be able to wear what we really like and not what we were supposed to due due utility.

**Comment by ispankzombiez** [comment]
I'd like to suggest a skill for the crops section that is "enhanced chance for seeds back from plants"

**Comment by alexandrelira99** [comment]
I'd like to suggest a skill for "single click to harvest & plant", similar to the "chop trees with one click"

**Comment by elowine** [comment]
Hello @adamhannigan  @TheDcol ,

After observing your recent developments, especially the numerous changes made to the tree about 15 days ago, I was wondering if it might be worth adopting a more gradual approach.

Why not start by releasing a bug-free tree, even if it’s not yet fully optimized, and then roll out regular updates on this topic? Here’s a concrete example of how this could work:

1. Day 1: The update is released, with an announcement stating that the skills might evolve over the next 1–2 weeks.
2. During this period, the SFL team analyzes the impact of the new skills, particularly on the P2P market.
3. Skills deemed too powerful or unbalanced are then adjusted. This process is repeated until a satisfactory balance is achieved.
4. With each rework or improvement, players are offered a free skill reset.

This approach could have a positive impact on players, as it demonstrates a commitment to regularly evolving the game. Many other games follow this strategy, and it seems to be well-received by their communities.

What do you think?
